### PR TITLE
[REF] Refactored column annotation flow store integration

### LINKED ISSUE #330: Refactor column annotation flow store integration
What we want to happen:
- User can update column description, data type, and it's standardized variable mapping

What should happen for this to be working with the new store:
- [x] Implement `userUpdatesColumnDescription` action and its unit tests
- [x] Implement `applyDataTypeToColumn` utility function and its unit tests
  -  contains the logic of determining fields based on the data type change and will be called inside the `userUpdatesColumnDataType` and `userUpdatesColumnStandardizedVariable` 
- [x] Implement `userUpdatesColumnDataType` action and its unit tests
- [x] Implement `userUpdatesColumnStandardizedVariable` and its unit tests
- [x] Implement `useDisabledStandardizedVariables` custom hook and its unit tests to determine if the standardized variable dropdown for a column should be disabled
  - this hook essentially replaces `mappedSingleColumnStandardizedVariable` and some of the logic related to disabling dropdowns that are in `ColumnAnnotationCard`
- [x] Update `ColumnAnnotation` to use `useColumns` and `useStandardizedVariables` selectors, handle logic for calling the store actions (this logic would be moved over from the child)
- [x] Update `ColumnAnnotationCard` to be as simple as possible, receive only the necessary data (just standardized variable labels and objects with ID, no more accessing store directly, etc) through props from the parent
- [x] Update `ColumnAnnotationCard` component test
- [x] Update `new` e2e test

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #330 

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [x] Checks pass
- [ ] If the PR changes the default [Neurobagel `config.json` or any of the term files](https://github.com/neurobagel/communities/tree/main/configs/Neurobagel), make sure their respective counterparts in [neurobagel/communities](https://github.com/neurobagel/communities) have been updated accordingly.

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

## Summary by Sourcery

Introduce a hook to determine disabled standardized variable options in the column annotation flow based on existing mappings and variable metadata.

New Features:
- Add useDisabledStandardizedVariables hook to compute disabled standardized variable labels based on column assignments and single-column constraints

Tests:
- Add unit tests for useDisabledStandardizedVariables covering no mappings, multi-column variables, single-column variables, invalid or null IDs, and empty standardizedVariables scenarios

## Summary by Sourcery

Refactor the column annotation flow by integrating new store actions and hooks to manage column description, data type, and standardized variable updates, introduce logic to disable already mapped single-column variables, and update components and tests to use the revamped FreshNewStore APIs.

New Features:
- Add useDisabledStandardizedVariables hook to disable single-column standardized variables that are already mapped
- Implement userUpdatesColumnDescription, userUpdatesColumnDataType, and userUpdatesColumnStandardizedVariable actions in FreshNewStore
- Introduce applyDataTypeToColumn utility to centralize data type transitions including levels and units

Enhancements:
- Refactor ColumnAnnotation and ColumnAnnotationCard to consume new store hooks and props, enabling conditional data type editing and disabled variable options
- Update datamodel to allow null values for descriptions, data types, and standardized variables
- Export raw FreshNewStore for testing purposes

Tests:
- Add unit tests for applyDataTypeToColumn covering categorical, continuous, and null cases
- Extend FreshNewStore tests for column description, data type switching, and standardized variable updates including multi-column measures
- Add tests for useDisabledStandardizedVariables hook scenarios
- Update Cypress E2E and component tests to validate new flows and disabled standardized variable behavior

### CODE REVIEW

File: src/components/ColumnAnnotationCard.tsx [hunk 1]
```diff
@@ -138,12 +102,10 @@ function ColumnAnnotationCard({
...
-              value={standardizedVariable?.label || ''}
-              onChange={handleStandardizedVariableChange}
-              options={Object.values(standardizedVariableOptions).map((value) => value.label)}
-              getOptionDisabled={(option) =>
-                mappedSingleColumnStandardizedVariables.map((sVar) => sVar.label).includes(option)
-              }
+              value={standardizedVariableLabel || ''}
+              onChange={(_, newValue) => onStandardizedVariableChange(id, newValue)}
+              options={standardizedVariableOptions}
+              getOptionDisabled={(option) => disabledStandardizedVariableLabels.has(option)}
```
Comment (surchs) [inline src/components/ColumnAnnotationCard.tsx]: from docs: https://mui.com/material-ui/api/autocomplete/#autocomplete-prop-getOptionDisabled `option` will have same shape as an individual item. so if `options` is an array of objects rather than strings, then here you will be able to filter for the "enabled" status of these items instead of looking at labels

File: src/hooks/useDisabledStandardizedVariables.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        // Only include if it's a single-column variable (not multi-column measure and can't have multiple columns)
+        const isSingleColumn =
+          variable.is_multi_column_measure !== true && variable.can_have_multiple_columns === false;
+
+        if (isSingleColumn) {
+          disabledLabels.add(variable.name);
+        }
+      }
+    }
+  });
```
Comment (surchs) [inline src/hooks/useDisabledStandardizedVariables.ts]: alternatively to what I suggested above to make this hook return an object with the required shape for ColumnAnnotation, it may be enough to just return SV IDs here (not labels). Then a second hook can use that hook and also the `useStandardizedVariables` hook to create the expected shape for ColumnAnnotations. That would allow for each hook to remain focused on a single thing. I'd say the important part is to not pass labels, but instead pass IDs

File: datamodel.ts [hunk 3]
```diff
@@ -18,12 +18,12 @@ interface Column {
   id: string;
   name: string;
   allValues: string[]; // because we want to show the datable preview
-  description?: string;
-  dataType?: DataType;
+  description?: string | null;
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply datamodel.ts]: I believe that's something that requires further discussion. Conceptually this change makes types clearer but given the current implementation across store, components, and tests it would require a sizable refactor in the code.

File: datamodel.ts [hunk 4]
```diff
@@ -18,12 +18,12 @@ interface Column {
...
+  dataType?: DataType | null;
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply datamodel.ts]: Semantically we treat `null` as the absence of value/unset state and including it in the `DataType` would be treating as a value.

File: src/hooks/useDisabledStandardizedVariables.ts (hunk 2 above)
Comment (surchs) [reply src/hooks/useDisabledStandardizedVariables.ts]: Makes sense, and we can revisit/refactor once there is a use case

File: src/hooks/useStandardizedVariableOptions.test.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        standardizedVariable: 'nb:NonExistent',
+      },
+      '2': {
+        id: '2',
+        name: 'participant_id',
+        allValues: [],
+        standardizedVariable: 'nb:ParticipantID',
+      },
+    });
+    mockedUseStandardizedVariables.mockReturnValue(mockFreshStandardizedVariables);
```
[the remark this answers is not in the document]
Comment (surchs) [reply src/hooks/useStandardizedVariableOptions.test.ts]: and if you can think of a theoretical case where this happens, please put it in an issue for reference / later

File: src/stores/FreshNewStore.test.ts [hunk 6]
```diff
@@ -521,6 +522,576 @@ describe('userUploadsDataDictionaryFile', () => {
...
+    act(() => {
+      result.current.actions.userUpdatesColumnDataType('0', DataType.categorical);
+    });
+
+    expect(result.current.columns['0'].dataType).toBe(DataType.categorical);
+    expect(result.current.columns['0'].units).toBeUndefined();
+    expect(result.current.columns['0'].levels).toBeDefined();
+  });
+
+  it('should clear dataType, levels, and units when dataType is null', async () => {
```
Comment (surchs) [inline src/stores/FreshNewStore.test.ts]:
```suggestion
  it('should clear dataType, levels, and units when dataType is set to null', async () => {
```

File: src/stores/FreshNewStore.test.ts [hunk 7]
```diff
@@ -521,6 +522,576 @@ describe('userUploadsDataDictionaryFile', () => {
...
+      result.current.actions.userUpdatesColumnStandardizedVariable('0', 'nb:Age');
+    });
+
+    expect(result.current.columns['0'].standardizedVariable).toBe('nb:Age');
+    expect(result.current.columns['0'].dataType).toBe(DataType.continuous);
+    expect(result.current.columns['0'].units).toBe('');
+    expect(result.current.columns['0'].levels).toBeUndefined();
+  });
+
+  it('should initialize isPartOf for multi-column measure variables', async () => {
```
Comment (surchs) [inline src/stores/FreshNewStore.test.ts]:
```suggestion
  it('should initialize isPartOf when column is mapped to Collection type variables', async () => {
```

File: src/utils/store-utils.ts [hunk 8]
```diff
@@ -210,6 +210,46 @@ export function parseTsvContent(content: string): { headers: string[]; data: str
   return { headers, data };
 }
 
+/**
+ * Applies data type changes to a column draft (for use with Immer produce).
+ * It mutates the draft directly.
+ */
```
Comment (surchs) [inline src/utils/store-utils.ts]: Thanks for the comment (missed this on first read), but I don't think that's the only way to make a util function work with immer. We should definitely find a way to not introduce util functions with side effects here - also makes for pretty awkward tests. I left a comment on the store hook that calls this on why I think we can just return the update here as a pure function and then handle the assembly in the store.

File: src/stores/FreshNewStore.test.ts (hunk 6 above)
Comment (rmanaem) [reply src/stores/FreshNewStore.test.ts]: I agree. We should also discuss structuring store unit tests as this file keeps growing with every new action.

Comment (rmanaem) [reply src/stores/FreshNewStore.test.ts]: https://github.com/neurobagel/annotation-tool/issues/350

File: src/utils/store-utils.test.ts [hunk 9]
```diff
@@ -720,188 +720,145 @@ describe('applyDataDictionaryToColumns', () => {
 });
 
 describe('applyDataTypeToColumn', () => {
-  it('should set dataType to categorical and initialize levels from allValues', () => {
-    const draft: {
+  const createColumn = (
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply src/utils/store-utils.test.ts]: No `beforeEach` is needed in this block. Each test creates its own fresh column via the `createColumn` helper, and there are no mocks or shared mutable state that would require resetting between cases.
